src/block_entity.zig

```diff
@@ -120,41 +131,15 @@ fn BlockEntityDataStorage(comptime side: enum {client, server}, T: type) type {
 			chunk.blockPosToEntityDataMapMutex.unlock();
 
 			const entry = entityNullable orelse {
-				std.log.warn("Couldn't remove entity data of block at position {}", .{pos});
+				std.log.err("Couldn't remove entity data of block at position {}", .{pos});
 				return;
 			};
 
 			const dataIndex = entry.value;
-			_ = storage.swapRemove(dataIndex);
-			if(dataIndex == storage.items.len) {
-				return;
-			}
-
-			const movedEntry = storage.items[dataIndex];
```

Comment:
Okay, cool, so what's the plan for unloading block entities when chunk they come from gets unloaded? Do you just iterate all the block entities whenever a chunk is unloaded?

Reply:
I'd just iterate over the chunk hashmap

Reply:
> Alright I guess, why did I remove it the way I did...?

To keep the list sparse, which is no longer needed due to using the SparseSet.